### PR TITLE
COMP: Simplify directory listing using vtksys::Directory

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: RigidAlignment.cxx [hunk 1]
```diff
@@ -111,24 +108,29 @@ int main(int argc, char* argv[])
...
+    if (vtksys::SystemTools::FileIsDirectory(filepath))
     {
-      string filename = entry->d_name;
-      if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
-      {
-        list.push_back(path + "/" + filename);
-      }
+      continue;
+    }
+    if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
```
Comment (Connor-Bowley) [inline RigidAlignment.cxx]: I don't think the logic here is actually correct (the original logic, which your change replicates).

`std::string::find_last_of` has the following semantics (from https://en.cppreference.com/w/cpp/string/basic_string/find_last_of)

> Finds the last character equal to **_one of_** characters in the given character sequence.

The current use of this is `if(! getListFile(landmark.c_str(), landmarkList, "fcsv"))` which makes it seem like we are looking for a `.fcsv` file, but in reality a `.mov` file would incorrectly match.

I think this needs to change to use something like `std::string::rfind` instead.

File: RigidAlignment.cxx [hunk 2]
```diff
@@ -111,24 +108,29 @@ int main(int argc, char* argv[])
...
+      list.push_back(filepath);
     }
-    closedir(dir);
-    sort(list.begin(), list.begin() + list.size());
-    return true;
-  } else {
-    cerr<<"The directory does not exist! "<<path<<endl;
-    return false;
   }
+  sort(list.begin(), list.begin() + list.size());
```
Comment (Connor-Bowley) [inline RigidAlignment.cxx]: If we are changing things anyway can we switch this to be `sort(list.begin(), list.end())` to be more idiomatic and clear we are sorting the whole list.

**Review by Connor-Bowley — CHANGES_REQUESTED** [review]

File: RigidAlignment.cxx (hunk 1 above)
Comment (Connor-Bowley) [reply RigidAlignment.cxx]: I have updated this to be correct (although it doesn't use rfind, as that is more inefficient than necessary).

**Review by Connor-Bowley — APPROVED** [review]

**Comment by jcfr** [comment]
Thanks for helping out with this 🙏

**Pull request merged**